test/signal_test.dart

```diff
@@ -0,0 +1,385 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:cli_script/cli_script.dart';
+import 'package:test/test.dart';
+
+import 'util.dart';
+
+void main() {
+  group('on Script', () {
+    test('interrupts with default signal', () async {
+      var script =
+          mainScript('await Future<void>.delayed(Duration(seconds: 100));');
+
+      expect(await script.kill(), true);
+
+      expect(script.done, throwsScriptException(-15));
+    });
+
+    test('interrupts with custom signal', () async {
+      var script =
+          mainScript('await Future<void>.delayed(Duration(seconds: 100));');
+
+      expect(await script.kill(ProcessSignal.sigint), true);
+      expect(script.done, throwsScriptException(-2));
+    });
+
+    test('can interrupt after Future.timeout', () async {
+      var script =
+          mainScript('await Future<void>.delayed(Duration(seconds: 100));');
+
+      script.done.timeout(Duration(milliseconds: 1), onTimeout: script.kill);
+      expect(script.done, throwsScriptException(-15));
+    });
+
+    test("can't be interrupt after it already exited", () async {
+      var script = mainScript('print("done!");');
+
+      expect(await script.output, 'done!');
+      expect(await script.kill(), false);
+    });
+
+    test("can be trapped without interrupting", () async {
+      var script = _watchSignalsAndExit();
+      var output = script.output;
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(await script.kill(), true);
+
+      expect(await output, 'SIGTERM\nbye!');
+      expect(await script.kill(), false);
+    });
+  });
+
+  group('on Script.capture', () {
+    test("does not interrupt by default", () async {
+      var script = Script.capture((_) async {
+        await Future<void>.delayed(Duration(seconds: 3));
+        print('done!');
+      });
+      var output = script.output;
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(await script.kill(), true);
+
+      expect(await output, 'done!');
+      expect(await script.kill(), false);
+    });
+
+    test("can be captured with onSignal", () async {
+      late Script innerScript;
+      var script = Script.capture((_) async {
+        innerScript = _watchSignalsAndExit();
```

Comment:
It's probably a good idea to avoid using physical subprocesses in these tests when it's not crucial to the behavior being tested, for performance if nothing else.